src/bundle/Resources/public/js/scripts/admin.limitation.pick.js

```diff
@@ -1,13 +1,159 @@
+        const fragment = doc.createDocumentFragment();
+
+        newlySelectedItems.forEach((location) => {
+            const locationId = location.id;
+            const container = doc.createElement('ul');
+            const renderedItem = tagTemplate.replace('{{ location_id }}', locationId);
+
+            container.insertAdjacentHTML('beforeend', renderedItem);
+
+            const listItemNode = container.querySelector('li');
+            const tagNode = listItemNode.querySelector(SELECTOR_EZ_TAG);
+
+            attachTagEventHandlers(limitationBtn, tagNode);
+            fragment.append(listItemNode);
+        });
+
+        tagsList.append(fragment);
+
+        setTagsBreadcrumbs(tagsList, newlySelectedItems);
+    };
+    const setTagsBreadcrumbs = (tagsList, newlySelectedItems) => {
+        const pathArraysWithoutRoot = newlySelectedItems.map(getLocationPathArray);
+
+        findLocationsByIdList(pathArraysWithoutRoot, (response) => {
+            const { operations } = response.BulkOperationResponse;
+
+            Object.entries(operations).forEach(([locationId, { content }]) => {
+                const viewData = JSON.parse(content);
+                const tag = tagsList.querySelector(`[data-location-id="${locationId}"]`);
+                const tagContent = tag.querySelector('.ez-tag__content');
+                const tagSpinner = tag.querySelector('.ez-tag__spinner');
+
+                tagContent.innerText = buildContentBreadcrumbs(viewData);
+
+                tagSpinner.hidden = true;
+                tagContent.hidden = false;
+            });
+        });
+    };
+    const getLocationPathArray = (location) => {
+        const { pathString } = location;
+        const pathArray = pathStringToPathArray(pathString);
+        const pathArrayWithoutRoot = removeRootLocation(pathArray);
+
+        return pathArrayWithoutRoot;
+    };
+    const handleTagRemove = (limitationBtn, tag) => {
+        const removedLocationId = tag.dataset.locationId;
+
+        removeLocationFromInput(limitationBtn, removedLocationId);
+        tag.remove();
+    };
+    const attachTagEventHandlers = (limitationBtn, tag) => {
+        const removeTagBtn = tag.querySelector('.ez-tag__remove-btn');
+
+        removeTagBtn.addEventListener('click', handleTagRemove.bind(null, limitationBtn, tag), false);
+    };
     const closeUDW = () => ReactDOM.unmountComponentAtNode(udwContainer);
-    const selectLocationsConfirm = (target, data) => {
-        const selectedItems = data.reduce((total, item) => total + `<li>${item.ContentInfo.Content.Name}</li>`, '');
+    const handleUdwConfirm = (limitationBtn, selectedItems) => {
+        const input = doc.querySelector(limitationBtn.dataset.locationInputSelector);
+        const alreadySelectedLocationsIds = input.value.split(IDS_SEPARATOR).map((idString) => parseInt(idString, 10));
+        const newlySelectedItems = selectedItems.filter((item) => !alreadySelectedLocationsIds.includes(item.id));
```

Comment:
I'm not sure if we want this, it might be confusing a little bit. But current solution is also a little bit confusing. 🤔 The better solution would be to preselect already selected items in UDW based on previous selection, but UDW AFAIK is not capable of doing this so it would require another implementing another feature.

For now, should we prevent selecting already selected locations in UDW or maybe, like now, allow to select any location and then exclude locations which were already selected? ping @SylvainGuittard @inakijv

Reply:
We have the same behavior in object relation list. You are not able to select in UDW already selected items. We should be consistent.

Reply:
The behaviour consistency is a key. I agree with Darek's solution.